Assets/_Scripts/Managers/CanvasManager.cs

```diff
@@ -0,0 +1,186 @@
+            startButton.onClick.AddListener(() => GameManager.Instance.ChangeScene(1));
+        }
+            
+
+        if (gameOverQuitButton || pauseQuitButton)
+        {
+            gameOverQuitButton.onClick.AddListener(Quit);
+            pauseQuitButton.onClick.AddListener(Quit);
+        }
+
+        if (mainMenuQuitButton)
+            mainMenuQuitButton.onClick.AddListener(Quit);
+
+        if (playAgainButton)
+        {
+            playAgainButton.onClick.AddListener(() => GameManager.Instance.RestartGame());
+            playAgainButton.onClick.AddListener(() => SetMenus(null, gameOverMenu));
+        }
+           
+
+        if (resumeButton)
+            resumeButton.onClick.AddListener(() => SetMenus(null, pauseMenu));
+
+        if (mainMenuButton)
+            mainMenuButton.onClick.AddListener(() => GameManager.Instance.ChangeScene(0));
+
+
+    }
+
+
+
+    //Sets menus from active to inactive.
+    void SetMenus(GameObject menuToActive, GameObject menuToInactive)
+    {
+        if (menuToActive)
+            menuToActive.SetActive(true);
+
+        if (menuToInactive)
+            menuToInactive.SetActive(false);
+    }
+
+
+
+    //Quits the game and stops playing in unity.
+    private void Quit()
+    {
+#if UNITY_EDITOR
+        EditorApplication.isPlaying = false;
+#else
+            Application.Quit();
+#endif
+    }
+
+
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (!pauseMenu) return;
+
+
+        if (Input.GetKeyDown(KeyCode.P))
+        {
+            pauseMenu.SetActive(!pauseMenu.activeSelf);
+
+        }
+        if (pauseMenu.activeInHierarchy)
+        {
+            Time.timeScale = 0;
+        }
+        else
+        {
+            Time.timeScale = 1;
+        }
+
+        if (GameManager.Instance.Lives == 0)
+        {
+            gameOverMenu.SetActive(true);
+        }
+
+        if (scoreText)
+            scoreText.text = GameManager.Instance.Score.ToString();
+
+
+        //sets lives images to the number of player lives.
+        float LivesImage = GameManager.Instance.Lives;
+
+        switch (LivesImage)
+        {
+            case 3:
+
+                LifeOne.enabled = true; 
+                LifeTwo.enabled = true; 
+                LifeThree.enabled = true;                
+                break;
+
+             case 2:
+
+                LifeTwo.enabled = true;
+                LifeOne.enabled = true;
+                LifeThree.enabled= false;
+                break;
+
+
+
+             case 1:
+
+                LifeOne.enabled = true;
+                LifeTwo.enabled = false;
+                LifeThree.enabled = false;
+
+                break;
+
+            case 0:
+
+                LifeOne.enabled = false;
+                LifeTwo.enabled= false;
+                LifeThree.enabled = false;
+                break;
+
+
+              default:
+              break;
+        }
```

Comment:
While this works, it is best to avoid code repetition and do this in a more readable and cleaner way using arrays.
Here's an idea of the suggestion:
```csharp
    // This will be exposed as a list of Images in the Unity editor.
    // Each image should be added as an element (in order: 1Life, 2Life, 3Life) to the list in Unity.
    [Header("Image")]
    public Image[] LifeImages;

   // ...

    // Update is called once per frame
    void Update()
    {
        // ...

        // Sets appropriate life images to display according to number of player lives remaining.
        for (int i = 0; i < LifeImages.Length; ++i)
        {
            LifeImages[i].enabled = i < GameManager.Instance.Lives;
        }

        // ...
    }
```

Rather than a long switch statement, this just goes through through each `LifeImage` and enables it or disables it depending on the number of lives that are left.

Even better, this doesn't need to be done on every frame update. It only needs to be done when the number of lives changes meaning when `GameManager.Instance.Lives` changes. You could hook this up by writing a function here `UpdateLifeHUD(lives)` that is called by the `GameManager` whenever the number of lives changes.
```csharp
public void UpdateLifeHUD(int lives)
{
    for (int i = 0; i < LifeImages.Length; ++i)
    {
        LifeImages[i].enabled = i < lives;
    }
}
```